components/bl-tour-guide-component/src/index.js

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useState } from 'react';
+
+import Joyride, { ACTIONS, STATUS } from './lib/react-joyride.min';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function TourGuideComponent({ component, eventHandlers, elRef }) {
+  const { onChange } = eventHandlers;
+  const {
+    classList, display, style, controlVisibility, controlLabel, options, continuous, autoStart, closeOnEscape,
+    closeOnClickOverlay, scrolling, locale, scrollOffset, scrollDuration, scrollToFirstStep, spotlightClicks,
+  } = component;
+
+  const {
+    progressVisibility, skipButtonVisibility, overlayVisibility, backButtonVisibility, closeButtonVisibility,
+  } = component;
+
+  const {
+    spotlightPadding, beaconSize, arrowColor, backgroundColor, overlayColor, primaryColor, textColor, width,
+  } = component;
```

Comment:
```suggestion
  const {
    classList, display, style, controlVisibility, controlLabel, options, continuous, autoStart, closeOnEscape,
    closeOnClickOverlay, scrolling, locale, scrollOffset, scrollDuration, scrollToFirstStep, spotlightClicks,
    progressVisibility, skipButtonVisibility, overlayVisibility, backButtonVisibility, closeButtonVisibility,
    spotlightPadding, beaconSize, arrowColor, backgroundColor, overlayColor, primaryColor, textColor, width,
  } = component;
```

Could we make use destruction once?